llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -139,6 +139,10 @@ bool AMDGPUInstructionSelector::selectCOPY(MachineInstr &I) const {
       return RBI.constrainGenericRegister(DstReg, *RC, *MRI);
     }
 
+    // Allow copy from physical register other than SCC to s1.
+    if (SrcReg.isPhysical() && SrcReg != AMDGPU::SCC)
```

Comment:
SrcReg != AMDGPU::SCC is redundant with the above scc check.This also early exists without constraining the result register.

You also cannot interpret a register without a type. You must take the type from the virtual register in the other position. This is exactly the same as in the regbankselect patch

Reply:
Added to the condition a check on type of DsgReg being i1.